File: java/src/main/java/br/com/summa/zx/StringBitOutputStream.java
/**
 * Zeta-Xi Code - by Einar Saukas
 *
 * See: https://github.com/einar-saukas/Zeta-Xi-Code
 */

package br.com.summa.zx;

public class StringBitOutputStream implements BitOutputStream {
    private final StringBuilder output = new StringBuilder();

    @Override
    public void writeBit(int bit) {
        output.append(bit > 0 ? '1' : '0');
    }

    public String result() {
        return output.toString();
    }
}

File: kotlin/src/main/kotlin/br/com/summa/zx/StringBitOutputStream.kt
/**
 * Zeta-Xi Code - by Einar Saukas
 *
 * See: https://github.com/einar-saukas/Zeta-Xi-Code
 */

package br.com.summa.zx

import java.lang.StringBuilder

class StringBitOutputStream : BitOutputStream {
    private val output = StringBuilder()

    override fun writeBit(bit: Int) { output.append(if (bit > 0) '1' else '0') }
    fun result() = output.toString()
}